### PR TITLE
feat: Remove deprecated policy action 

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
https://github.com/terraform-aws-modules/terraform-aws-sns/issues/46

## Motivation and Context
https://github.com/terraform-aws-modules/terraform-aws-sns/issues/46

## Breaking Changes
AWS provider v5. 

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/versions.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.40"
+      version = ">= 5.0"
```
Comment (antonbabenko) [inline examples/complete/versions.tf:7]: Why is this change necessary? I don't think you introduce anything backward incompatible with v4.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: examples/complete/versions.tf (hunk 1 above)
Comment (magreenbaum) [reply examples/complete/versions.tf]: You're right, it's not necessary, sorry. I've reverted this v5 upgrade.

**Comment by antonbabenko** [comment]
This PR is included in version 5.4.0 :tada:
